applications/ecallmgr/src/ecallmgr_fs_authz.erl

```diff
@@ -181,8 +181,24 @@ authz_response(JObj, Props, CallId, Node) ->
                 'true' -> authorize_account(JObj, Props, CallId, Node);
                 'false' ->
                     %% Set the following CCVs so that we can see why the call was barred in CDRs
+
+                    %% set Account-ID and Reseller-ID so CDRs can be saved for this call
+                    %% (in case the call is inbound from carrier)
+                    AccountId = kz_json:get_value(<<"Account-ID">>
+                                                 ,JObj
+                                                 ,props:get_value(?GET_CCV(<<"Account-ID">>), Props)
+                                                 ),
+                    ResellerId = kz_json:get_value(<<"Reseller-ID">>
+                                                  ,JObj
+                                                  ,props:get_value(?GET_CCV(<<"Account-ID">>)
+                                                                  ,Props
+                                                                  ,kz_services_reseller:get_id(AccountId)
```

Comment:
that's a lot of defaults.
you're making ecallmgr aware of kz_services which is not good, ecallmgr should deal with props / json.
i think you can set the variables that come out of j5 either authorized or not.